File: Algos/Stacks/BalancedBrackets.py
 class Stack():
    def __init__(self):
        self.items = []

    def is_empty(self):
        return self.items == []

    def push(self, item):
        self.items.append(item)

    def pop(self):
        return self.items.pop()

    def peek(self):
        last = len(self.items) - 1
        return self.items[last]

    def size(self):
        return len(self.items)

 def par_checker(symbol_string):
      s = Stack()
       for symbol in symbol_string:
            if symbol == "(":
                s.push(symbol)
            else:
                if s.is_empty():
                    return False
                else:
                    s.pop()

        return s.is_empty()

File: AlgoDaily/week-three/balancedBrackets.js
const validateSymbols = (str) => {
  // implement this method
  const map = {
    "(": ")",
    "{": "}",
    "[": "]",
  };

  const stack = [];
  for (let val of str) {
    if (val === "{" || val === "[" || val === "(") {
      stack.push(val);
    } else {
      if (val === map[stack[stack.length - 1]]) {
        stack.pop();
      } else {
        return false;
      }
    }
  }
  return stack.length === 0 ? true : false;
};
